src/storage/mysql_catalog.cpp

```diff
@@ -325,32 +349,116 @@ string MySQLCatalog::GetConnectionString(ClientContext &context, const string &a
 		// provided
 		secret_name = "__default_mysql";
 	}
+
 	auto secret_entry = GetSecret(context, secret_name);
-	auto connection_string = attach_path;
+	string connection_string = attach_path;
+	StringUtil::Trim(connection_string);
 
 	// if the connection string is a URI, try and convert it
 	TryConvertURI(connection_string);
 
 	if (secret_entry) {
 		// secret found - read data
 		const auto &kv_secret = dynamic_cast<const KeyValueSecret &>(*secret_entry->secret);
-		string new_connection_info;
-
-		new_connection_info += AddConnectionOption(kv_secret, "user");
-		new_connection_info += AddConnectionOption(kv_secret, "password");
-		new_connection_info += AddConnectionOption(kv_secret, "host");
-		new_connection_info += AddConnectionOption(kv_secret, "port");
-		new_connection_info += AddConnectionOption(kv_secret, "database");
-		new_connection_info += AddConnectionOption(kv_secret, "socket");
-		new_connection_info += AddConnectionOption(kv_secret, "ssl_mode");
-		new_connection_info += AddConnectionOption(kv_secret, "ssl_ca");
-		new_connection_info += AddConnectionOption(kv_secret, "ssl_capath");
-		new_connection_info += AddConnectionOption(kv_secret, "ssl_cert");
-		new_connection_info += AddConnectionOption(kv_secret, "ssl_cipher");
-		new_connection_info += AddConnectionOption(kv_secret, "ssl_crl");
-		new_connection_info += AddConnectionOption(kv_secret, "ssl_crlpath");
-		new_connection_info += AddConnectionOption(kv_secret, "ssl_key");
-		connection_string = new_connection_info + connection_string;
+
+		// If connection_string is empty, we can just use the parameters from the secret
+		if (connection_string.empty()) {
+			string new_connection_info;
+
+			new_connection_info += AddConnectionOption(kv_secret, "user");
```

Comment:
This special case here for when no connection string is present seems unnecessary - we can just leave `existing_params` empty which will have the same effect.